streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableTransformValuesTest.java

```diff
@@ -0,0 +1,553 @@
+        props.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9091");
+        props.setProperty(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getAbsolutePath());
+        props.setProperty(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass().getName());
+        props.setProperty(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass().getName());
+        return props;
+    }
+
+    private static void throwIfStoresNotAvailable(final ProcessorContext context,
+                                                  final List<String> expectedStoredNames) {
+        final List<String> missing = new ArrayList<>();
+
+        for (String storedName : expectedStoredNames) {
+            if (context.getStateStore(storedName) == null) {
+                missing.add(storedName);
+            }
+        }
+
+        if (!missing.isEmpty()) {
+            throw new AssertionError("State stores are not accessible: " + missing);
+        }
+    }
+
+    public static class ExclamationValueTransformerSupplier implements ValueTransformerWithKeySupplier<Object, String, String> {
+        private final List<String> expectedStoredNames;
+
+        ExclamationValueTransformerSupplier(final String... expectedStoreNames) {
+            this.expectedStoredNames = Arrays.asList(expectedStoreNames);
+        }
+
+        @Override
+        public ExclamationValueTransformer get() {
+            return new ExclamationValueTransformer(expectedStoredNames);
+        }
+    }
+
+    public static class ExclamationValueTransformer implements ValueTransformerWithKey<Object, String, String> {
+        private final List<String> expectedStoredNames;
+
+        ExclamationValueTransformer(final List<String> expectedStoredNames) {
+            this.expectedStoredNames = expectedStoredNames;
+        }
+
+        @Override
+        public void init(final ProcessorContext context) {
+            throwIfStoresNotAvailable(context, expectedStoredNames);
+        }
+
+        @Override
+        public String transform(final Object readOnlyKey, final String value) {
+            return readOnlyKey.toString() + "->" + value + "!";
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    private static class NullSupplier implements ValueTransformerWithKeySupplier<String, String, String> {
+        @Override
+        public ValueTransformerWithKey<String, String, String> get() {
+            return null;
+        }
+    }
+
+    private static class StatefulTransformerSupplier implements ValueTransformerWithKeySupplier<String, String, Integer> {
+        @Override
+        public ValueTransformerWithKey<String, String, Integer> get() {
+            return new StatefulTransformer();
+        }
+    }
+
+    private static class StatefulTransformer implements ValueTransformerWithKey<String, String, Integer> {
```

Comment:
maybe rename s.th. it describes what it does?

Reply:
What it does isn't as important as the fact its stateful vs stateless IMHO.